storages/backends/azure_storage.py

```diff
@@ -1,42 +1,123 @@
-from datetime import datetime
+from datetime import datetime, timedelta
+from azure.storage.blob import BlobBlock
 import os.path
 import mimetypes
-import time
-from time import mktime
-
-from django.core.files.base import ContentFile
-from django.core.exceptions import ImproperlyConfigured
 from django.core.files.storage import Storage
 from django.utils.deconstruct import deconstructible
-
-try:
-    import azure  # noqa
-except ImportError:
-    raise ImproperlyConfigured(
-        "Could not load Azure bindings. "
-        "See https://github.com/WindowsAzure/azure-sdk-for-python")
-
-try:
-    # azure-storage 0.20.0
-    from azure.storage.blob.blobservice import BlobService
-    from azure.common import AzureMissingResourceHttpError
-except ImportError:
-    from azure.storage import BlobService
-    from azure import WindowsAzureMissingResourceError as AzureMissingResourceHttpError
-
+from azure.storage import CloudStorageAccount
 from storages.utils import setting
+from tempfile import SpooledTemporaryFile
+from django.core.files.base import File
+from django.utils.encoding import force_bytes
+
+from azure.common import AzureMissingResourceHttpError
+from azure.storage.blob import ContentSettings
+import base64
+from django.utils.six.moves import urllib
 
 
 def clean_name(name):
     return os.path.normpath(name).replace("\\", "/")
 
 
+def pad_left(n, width, pad="0"):
+    return ((pad * width) + str(n))[-width:]
+
+
+@deconstructible
+class AzureStorageFile(File):
+
+    def __init__(self, name, mode, storage):
+        self._name = name
```

Comment:
Does `_name`  have to be private (instead of `name`) in this case?

django.core.files.File objects call it `name`, which means the `__str__` method throws an error:

 ```
>>> from storages.backends.azure_storage import AzureStorageFile
>>> print(AzureStorageFile('file.csv', 'r', None))
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm 2017.2.3\helpers\pydev\_pydevd_bundle\pydevd_exec2.py", line 3, in Exec
    exec(exp, global_vars, local_vars)
  File "<input>", line 1, in <module>
  File "C:\Users\H193143\ntl_loader\venv\lib\site-packages\django\core\files\base.py", line 29, in __repr__
    return force_str("<%s: %s>" % (self.__class__.__name__, self or "None"))
  File "C:\Users\H193143\ntl_loader\venv\lib\site-packages\django\core\files\base.py", line 32, in __bool__
    return bool(self.name)
AttributeError: 'AzureStorageFile' object has no attribute 'name'
```

Overall, great stuff!